### PR TITLE
Adds Disclaimers on Challenges for CSS ID Selector and Inline Styling

### LINKED ISSUE #16874: Using id for styling is an anti-pattern
#### Challenge Name
Override Class Declarations by Styling ID Attributes

#### Issue Description
The above lesson as well as an earlier lesson in the current series, teaches students to use an id as a selector for styling elements. I personally feel that this should not be done.

While there can certainly be a lesson that explains that it is possible. I feel it should be framed with an understanding that it is a practice that should never be employed in projects. 

Instead of going over the reasons for this, it is probably best to link to the css-lint documentation. https://github.com/CSSLint/csslint/wiki/Disallow-ids-in-selectors

### PR DESCRIPTION
**Pull request by amity**
Added disclaimers on challenges requiring ID selectors and inline styling to indicate that these are not best practices.

- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
- [x] My article does not contain shortened URLs or affiliate links.

Closes #16874

I tried to keep the language on the disclaimer somewhat similar to that in the guide at https://github.com/freeCodeCamp/freeCodeCamp/blob/master/guide/english/css/id-selector/index.md

### CODE REVIEW

**Review by ezioda004 — CHANGES_REQUESTED** [review]
Hey @soph-iest,

Thank you for opening this PR.

I dont think we should add the same note to 3 different challenges. Adding to one of them is fine.
Particularly adding it to [this](https://learn.freecodecamp.org/responsive-web-design/basic-css/use-an-id-attribute-to-style-an-element) challenge would be better. 

The note should be added in the "Description" section.
 
Finally, note should be small, preferable 1-2 line. Can you improve the verbiage? 

Thank you.

**Comment by soph-iest** [comment]
Makes sense to me. How's the new wording look to you?

File: curriculum/challenges/english/01-responsive-web-design/basic-css/use-an-id-attribute-to-style-an-element.english.md [hunk 1]
```diff
@@ -12,6 +12,8 @@ However, an <code>id</code> is not reusable and should only be applied to one el
 Here's an example of how you can take your element with the <code>id</code> attribute of <code>cat-photo-element</code> and give it the background color of green. In your <code>style</code> element:
 <blockquote>#cat-photo-element {<br>&nbsp;&nbsp;background-color: green;<br>}</blockquote>
 Note that inside your <code>style</code> element, you always reference classes by putting a <code>.</code> in front of their names. You always reference ids by putting a <code>#</code> in front of their names.
+  
+One thing to note is that using ID selectors to style elements should be avoided. Using IDs for styling individual elements loses out on much of the functionality of CSS sheets, and because they have such high specificity, they override most other selectors for styling.
```
Comment (ezioda004) [inline curriculum/challenges/english/01-responsive-web-design/basic-css/use-an-id-attribute-to-style-an-element.english.md]: Can we be a bit more concise here? We only need to mention why IDs should be avoided.
Something like:

>**Note:**
ID selectors to style elements should be avoided. Due to their high specificity, they can override class and tag selectors.

Comment (soph-iest) [reply curriculum/challenges/english/01-responsive-web-design/basic-css/use-an-id-attribute-to-style-an-element.english.md]: Sure, sounds good.

**Comment by soph-iest** [comment]
Done -- thanks for writing that up.

**Comment by QuincyLarson** [comment]
@soph-iest That's better - could you make it more concise and remove the passive voice from it? If we can get it down to just one or two short sentences, that would be best.

**Comment by lasjorg** [comment]
How about this:

It is advisable not to use ids for styling. Styles should be easily overridden and extended, making your CSS more maintainable. The higher weight of importance given to the id selector can make your CSS less maintainable. You should prefer using classes for styling.

Or just:

It is advisable not to use ids for styling. The higher weight of importance given to the id selector can make CSS less maintainable.

---

Personally, I suggest this challenge be rewritten and brought "in-line" with the challenge on the class selector.

**Comment by lasjorg** [comment]
@ezioda004 Is one of the two suggestions I gave acceptable? I'm rewriting the challenge and would like to know if I can use one of them. The first one can also be a bit shorter by removing the mention of using classes.

**Comment by RandellDawson** [comment]
@ezioda004 What do you think about @lasjorg's suggestion?

**Comment by QuincyLarson** [comment]
We haven't heard from @soph-iest for a couple months so I'm closing this as stale.

**Comment by soph-iest** [comment]
Apologies, I've been very busy and haven't been monitoring my GitHub very closely. 

I personally prefer the former of @lasjorg 's suggestions, but I'm good with either.

**Pull request closed without merging**